src/core/EVM/EVMStepExecutor.ts

```diff
@@ -87,9 +86,13 @@ export class EVMStepExecutor extends BaseStepExecutor {
         },
       })
       this.statusManager.updateExecution(step, 'FAILED')
-      throw new TransactionError(
-        LiFiErrorCode.WalletChangedDuringExecution,
-        errorMessage
+      throw await parseEVMStepErrors(
+        new TransactionError(
+          LiFiErrorCode.WalletChangedDuringExecution,
+          errorMessage
+        ),
+        step,
+        process
```

Comment:
Do you think it makes sense to parse it here, as this code is usually wrapped inside another try-catch, and we will end up parsing it twice? Also, I think we should only parse errors before throwing them to the top as a last step.

Reply:
The checkWallet function isn't always used in a part of the executeStep method that isn't cover by parseEVMStep Errors in the try catch there. I think if we look to simplify the executeStep method future we could look to reduce the use of parseEVMStepErrors then.